src/checkers/inference/model/ComparisonVariableSlot.java

```diff
@@ -0,0 +1,41 @@
+package checkers.inference.model;
+
+/**
+ * ComparisonVariableSlot represent the left side refinement of an comparison operation between two other
+ * {@link VariableSlot}s. 
+ * e.g., for a comparison constraint c := x &lt y, the comparison variable slot c is the refined value of
+ * x where x < y is always when x = c.
+ * Note that this slot is serialized identically to a {@link VariableSlot}.
```

Comment:
Is this really true? This calls an overloaded `serialize` method and I don't see that it has to be the same as for `VariableSlot`.